File: 0008-string-to-integer-atoi/0008-string-to-integer-atoi.py
class Solution(object):
    def myAtoi(self, s):
        """
        :type s: str
        :rtype: int
        """
        s=s.strip()
        if(len(s)==0):
            return 0
        res=0
        i=0
        neg=False
        if(s[i]=='-' or s[i]=='+'):
            neg=(s[i]=='-')
            i=i+1
        while(i<len(s) and s[i].isdigit()):
            res=res*10+int(s[i])
            i=i+1
        if(neg):
            res=res*-1
        maxi=2**31-1
        mini=-(2**31)
        if(res>=maxi):
            return maxi
        elif(res<=mini):
            return mini
        else:
            return res

File: 0008-string-to-integer-atoi/0008-string-to-integer-atoi.cpp
class Solution {
public:
    int myAtoi(string s) {
        int p=0;
        bool minus=false;
        int maxi=INT_MAX;
        int mini=INT_MIN;
        int n=s.size();
        while(p<n && s[p]==' ')
        {
            p+=1;
        }
        if(p<n && s[p]=='+' || s[p]=='-')
        {
            minus=(s[p]=='-');
            p+=1;
        }
        int res=0;
        while(p<n && isdigit(s[p]))
        {
            int digit=s[p]-'0';
            if(res>((maxi-digit)/10))
            {
                if(minus) return mini;
                return maxi;
            }
             res=res*10+digit;
            p+=1;
        }
        if(minus) res=res*-1;
        if(res<=mini) return mini;
        return res;
    }
};